core/performance_test/peakflops/02_kokkos_simple_view.cpp

```diff
@@ -0,0 +1,189 @@
+#include <Kokkos_Core.hpp>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <gtest/gtest.h>
+#include "common.h"
+//#include "rdtsc.h"
+
+typedef Kokkos::View<float[VECLENTH],Kokkos::MemoryTraits<Kokkos::Restrict> > view_type;
+
+/*
+void __attribute__ ((noinline)) inita(const view_type & __restrict__  a,const view_type & __restrict__  x0,const view_type & __restrict__   x1,const view_type & __restrict__   x2,const view_type & __restrict__  c){
+    asm volatile ("# ax+c loop begin");
+    for(int l=1;l<10000;l++){
+#pragma omp simd
+        for(int j=0; j<8; j++){
+            x1(j) =a(j)*x1(j) +c(j);
+            x2(j) =a(j)*x2(j) +c(j);
+            x0(j) =a(j)*x0(j) +c(j);
+        }
+    }
+    asm volatile ("# ax+c loop end");
+}
+
+*/
+
+
+struct data{
+    float vec[VECLENTH];
+};
+
+//struct data * axpy_10(struct data *restrict a, struct data *restrict x0, struct data *restrict x1, struct data *restrict x2, struct data *restrict x3, struct data *restrict x4, struct data *restrict x5, struct data *restrict x6, struct data *restrict x7, struct data *restrict x8, struct data *restrict x9,struct data *restrict c, long n) {
+//struct data * axpy_10(struct data *__restrict__ a, struct data *__restrict__ x0, struct data *__restrict__ x1, struct data *__restrict__ x2, struct data *__restrict__ x3, struct data *__restrict__ x4, struct data *__restrict__ x5, struct data *__restrict__ x6, struct data *__restrict__ x7, struct data *__restrict__ x8, struct data *__restrict__ x9,struct data *__restrict__ c, long n) {
+
+view_type //__attribute__ ((noinline))
+axpy_10(
+        const view_type & __restrict__  a,
```

Comment:
Does using `__restrict__` here matter when they type is a `Kokkos::View`?

Reply:
It certainly doesn't hurt (and, I think, has different semantics). I'm not sure if it's performance critical though